### DISCUSSION TITLE
show internal and external links differently?

### DISCUSSION THREAD

**Discussion by X-Wei** [zim-desktop-wiki/zim-desktop-wiki#2553] [Ideas]
In zim a link can point to **another note** or point to some **external URL**. Currently both are rendered the same as blue underlined text.But it would be really nice to be able to show them differently. 

For example: internal links are green and external links are blue. The `internal_link`'s style shall be configurable via `styles.conf`.

Not sure if this can be achieved via a plugin, if it's possible, could anyone provide some pointers?

**Comment by introt** [comment]
There's an open ticket for this:  #197

As for pointers, you're going to want to start with modifying the TextBuffer to support multiple link styles, then look into how hrefs are handled to tell apart internal and external links. Do note that the docs can be a little out of date in certain places, eg. the TextBuffer docs suggest that it doesn't support overlapping italics and bold:

https://github.com/zim-desktop-wiki/zim-desktop-wiki/blob/70355fa66c9b6abfbb15aec2675cbae55a6153fe/zim/gui/pageview/textbuffer.py#L162-L199

**Comment by X-Wei** [comment]
Thanks a lot for the pointer! 

turns out it's not so difficult, I've created https://github.com/zim-desktop-wiki/zim-desktop-wiki/pull/2555
